### ISSUE TITLE
Issue: Segentation fault on run

### ISSUE THREAD

**Issue by DusterTheFirst** [nektos/act#737]
## System information
https://termbin.com/4pcq
Docker version 20.10.7, build f0df350
act version 0.2.23

## Expected behaviour

Act produces a useful error message or anything other than a panic

## Actual behaviour

act panics

## Workflow and/or repository

<details>
  <summary>workflow</summary>

```yml
name: Verify KiCAD Files
on: push
jobs:
  footprints:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
```

</details>

## Steps to reproduce
Install act through provided script. Run `act`.

## `act` output

<details>
  <summary>Log</summary>

```none
DEBU[0000] Loading environment from /home/dusterthefirst/Documents/kicad_libraries/Dusterthefirst-kicad/.env 
DEBU[0000] Loading secrets from /home/dusterthefirst/Documents/kicad_libraries/Dusterthefirst-kicad/.secrets 
DEBU[0000] Loading workflows from '/home/dusterthefirst/Documents/kicad_libraries/Dusterthefirst-kicad/.github/workflows' 
DEBU[0000] Loading workflows recursively                
DEBU[0000] Found workflow 'kicad-verify.yml' in '/home/dusterthefirst/Documents/kicad_libraries/Dusterthefirst-kicad/.github/workflows/kicad-verify.yml' 
DEBU[0000] Reading workflow '/home/dusterthefirst/Documents/kicad_libraries/Dusterthefirst-kicad/.github/workflows/kicad-verify.yml' 
DEBU[0000] Correcting if statements '/home/dusterthefirst/Documents/kicad_libraries/Dusterthefirst-kicad/.github/workflows/kicad-verify.yml' 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x8b9778]

goroutine 1 [running]:
github.com/nektos/act/pkg/model.FixIfStatement(0xc0004dc200, 0x6d, 0x200, 0xc0004d8000, 0x200, 0x0)
        /home/runner/work/act/act/pkg/model/planner.go:77 +0x198
github.com/nektos/act/pkg/model.NewWorkflowPlanner(0xc0004d20c0, 0x55, 0x0, 0xc0004d20c0, 0x55, 0x1, 0x1)
        /home/runner/work/act/act/pkg/model/planner.go:184 +0x6e6
github.com/nektos/act/cmd.newRunCommand.func1(0xc000492000, 0xc00048e0b0, 0x0, 0x1, 0x0, 0x0)
        /home/runner/work/act/act/cmd/root.go:173 +0x405
github.com/spf13/cobra.(*Command).execute(0xc000492000, 0xc00048e090, 0x1, 0x1, 0xc000492000, 0xc00048e090)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000492000, 0x1, 0x1, 0xf)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/nektos/act/cmd.Execute(0xe70fe0, 0xc0003f6600, 0xe4d8bc, 0x6)
        /home/runner/work/act/act/cmd/root.go:71 +0x100c
main.main()
        /home/runner/work/act/act/main.go:33 +0x175
```

</details>

**Comment by rainabba** [comment]
I'm seeing the same as I try to use act for the first time. 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x71 pc=0xa57e6e]

goroutine 74 [running]:
github.com/docker/docker/client.(*Client).getAPIPath(0x12653a0?, {0xd952b0?, 0xc000712a80?}, {0xc0004151a0, 0x14}, 0x14?)
        /home/michael/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/docker/docker@v20.10.14+incompatible/client/client.go:183 +0x4e
github.com/docker/docker/client.(*Client).sendRequest(0xc656a2?, {0xd952b0, 0xc000712a80}, {0xc65072, 0x3}, {0xc0004151a0?, 0x203000?}, 0x203000?, {0x0, 0x0}, ...)
        /home/michael/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/docker/docker@v20.10.14+incompatible/client/request.go:109 +0x78
github.com/docker/docker/client.(*Client).get(...)
        /home/michael/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/docker/docker@v20.10.14+incompatible/client/request.go:37
github.com/docker/docker/client.(*Client).CopyFromContainer(0x75?, {0xd952b0, 0xc000712a80}, {0x0, 0x0}, {0xc000049580, 0x75})
        /home/michael/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/docker/docker@v20.10.14+incompatible/client/container_copy.go:68 +0x1fe
github.com/nektos/act/pkg/container.(*containerReference).GetContainerArchive(0xc000415578?, {0xd952b0?, 0xc000712a80?}, {0xc000049580?, 0x450e57?})
        /tmp/act-20220426-721-neyxz8/pkg/container/docker_run.go:164 +0x59
github.com/nektos/act/pkg/runner.(*stepActionLocal).main.func1.1.1({0xc68e43?, 0x8?})
        /tmp/act-20220426-721-neyxz8/pkg/runner/step_action_local.go:41 +0xbd
github.com/nektos/act/pkg/runner.readActionImpl(0xc0001bfa40, {0xc00024a310, 0x6a}, {0x0, 0x0}, 0xc000216810, 0xcc1cf0)
        /tmp/act-20220426-721-neyxz8/pkg/runner/action.go:40 +0x9c
github.com/nektos/act/pkg/runner.(*stepActionLocal).main.func1({0xd952b0, 0xc000712a80})
        /tmp/act-20220426-721-neyxz8/pkg/runner/step_action_local.go:53 +0x13b
github.com/nektos/act/pkg/runner.runStepExecutor.func1({0xd952b0, 0xc000712a80})
        /tmp/act-20220426-721-neyxz8/pkg/runner/step.go:56 +0x363
github.com/nektos/act/pkg/runner.newJobExecutor.func2.1(0xd952b0?, {0xd952b0, 0xc000712a80})
        /tmp/act-20220426-721-neyxz8/pkg/runner/job_executor.go:58 +0x36
github.com/nektos/act/pkg/runner.newJobExecutor.func2({0xd952b0?, 0xc000474690?})
        /tmp/act-20220426-721-neyxz8/pkg/runner/job_executor.go:67 +0x8d
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:143 +0xe4
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Then.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:131 +0x34
github.com/nektos/act/pkg/common.Executor.Finally.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:177 +0x34
github.com/nektos/act/pkg/common.Executor.Finally.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:177 +0x34
github.com/nektos/act/pkg/runner.(*RunContext).Executor.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/runner/run_context.go:305 +0x7e
github.com/nektos/act/pkg/common.Executor.Finally.func1({0xd952b0, 0xc000474690})
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:177 +0x34
github.com/nektos/act/pkg/runner.(*runnerImpl).NewPlanExecutor.func1.1({0xd952b0, 0xc000474510})
        /tmp/act-20220426-721-neyxz8/pkg/runner/runner.go:179 +0x2f2
github.com/nektos/act/pkg/common.NewParallelExecutor.func1.1(0x0?, 0x0?)
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:102 +0x5c
created by github.com/nektos/act/pkg/common.NewParallelExecutor.func1
        /tmp/act-20220426-721-neyxz8/pkg/common/executor.go:100 +0x89
```

**Comment by rainabba** [comment]
Related? https://github.com/nektos/act/issues/438

**Comment by ChristopherHX** [comment]
This issue must have been fixed by https://github.com/nektos/act/commit/ff13844b8643abaff9daa47a5bc96aa91fddcf5e included in version v0.2.26.
The function which caused the crash no longer exists.

The provided workflow doesn't seem to contain any relevant content for the crash.

@rainabba Please open a new issue, your log is not related to this issue.
